odl/trafos/wavelet.py

```diff
@@ -21,9 +21,836 @@
 from __future__ import print_function, division, absolute_import
 from future import standard_library
 standard_library.install_aliases()
+from builtins import range, str, super
 
 # External
+import numpy as np
 
 # Internal
+from odl.discr.lp_discr import DiscreteLp
+from odl.operator.operator import Operator
 
-__all__ = ()
+try:
+    import pywt
+    WAVELET_AVAILABLE = True
+except ImportError:
+    WAVELET_AVAILABLE = False
+
+__all__ = ('DiscreteWaveletTrafo', 'DiscreteWaveletTrafoAdjoint',
+           'DiscreteWaveletTrafoInverse', 'WAVELET_AVAILABLE')
+
+
+_SUPPORTED_IMPL = ('pywt',)
+
+
+def list_of_coeff_sizes(shape, nscale, wbasis, mode):
```

Comment:
I'm not a friend of prepositions like "of", "by", "using" etc. in function names. An exception is "from", which is the standard preposition for alternative initializers, e.g. `fromiter`. In 99 % of the other cases, the same can be expressed without them. Here: `coeff_size_list`.